File: bin/plugins/kde.py
import subprocess
from bin import config


def switchToLight():
    kde_theme = config.getKdeLightTheme()
    # uses a kde api to switch to a light theme
    print(kde_theme)
    subprocess.run(["lookandfeeltool", "-a", kde_theme])


def switchToDark():

    kde_theme = config.getKdeDarkTheme()
    # uses a kde api to switch to a dark theme
    print(kde_theme)
    subprocess.run(["lookandfeeltool", "-a", kde_theme])

File: lib/plugins/kde.dart
import 'package:process_run/shell.dart';

class KDE {
  KDE();

  var shell = Shell();

  List<String> getThemes() {
    return [
      'org.kde.breezetwilight.desktop',
      'org.kde.breeze.desktop',
      'org.kde.breezedark.desktop'
    ];
  }

  void switchToDark() async {
    String theme = 'org.kde.breezedark.desktop';
    await shell.run('lookandfeeltool -a $theme');
  }

  void switchToLight() async {
    print('called');
    String theme = 'org.kde.breeze.desktop';
    await shell.run('lookandfeeltool -a $theme');
  }
}
